### PR TITLE
Implement variant DB2 improving the SECR

### PR DESCRIPTION
**Pull request by damerell**
I have not implemented this as two separate options since AFAIK anyone
who uses them uses them together, and '25 has _quite a lot_ of options
already.

### CODE REVIEW

**Comment by damerell** [comment]
I should perhaps have said that I am a Perl-writing sysadmin who knew no Ruby a week ago, so I expect you will need to correct me about some of what I have done.

File: lib/engine/game/g_1825/entities.rb [hunk 1]
```diff
@@ -554,6 +555,14 @@ def game_corporations
             gwr[:tokens] = [0, 0, 40, 100, 100, 100, 100]
             gwr[:coordinates] = %w[V14 Y7]
           end
+          # Modify SECR / LBSC in variant DB2
+          if @optional_rules.include?(:db2)
+            secr = corps.find { |corp| corp[:sym] == 'SECR' }
+            secr[:abilities] << { type: 'blocks_hexes', owner_type: nil, hexes: ['W21'] }
+            PAR_BY_CORPORATION['SECR'] = 67
+            lbsc = corps.find { |corp| corp[:sym] == 'LBSC' }
+            lbsc[:abilities] = nil
```
Comment (roseundy) [inline lib/engine/game/g_1825/entities.rb:564]: Shouldn't this be [] instead of nil ?

File: lib/engine/game/g_1825/meta.rb [hunk 2]
```diff
@@ -175,6 +181,8 @@ def self.check_options(options, num_players)
           regionals[2] = true if optional_rules.include?(:r2)
           regionals[3] = true if optional_rules.include?(:r3)
 
+          dbvariants[2] = true if optional_rules.include?(:db2)
```
Comment (roseundy) [inline lib/engine/game/g_1825/meta.rb:184]: Is dbvariants ever used? Why do you need this?

File: lib/engine/game/g_1825/entities.rb (hunk 1 above)
Comment (damerell) [reply lib/engine/game/g_1825/entities.rb]: Corrected.

File: lib/engine/game/g_1825/meta.rb (hunk 2 above)
Comment (damerell) [reply lib/engine/game/g_1825/meta.rb]: My mistake. I started thinking of a scheme like regionals[] etc but the Dave Berry variants don't really have much in common other than the author.

**Comment by roseundy** [comment]
Actually, I believe Ruby was inspired in part by Perl.

**Comment by damerell** [comment]
So do I, but (alas) being a mediocre Perl programmer doesn't help that much. :-/

**Pull request merged**